api-inference-community/docker_images/fasttext/app/pipelines/text_classification.py

```diff
@@ -0,0 +1,29 @@
+from typing import Dict, List
+
+from app.pipelines import Pipeline
+
+
+class TextClassificationPipeline(Pipeline):
+    def __init__(
+        self,
+        model_id: str,
+    ):
+        super().__init__(model_id)
+
+    def __call__(self, inputs: str) -> List[Dict[str, float]]:
+        """
+        Args:
+            inputs (:obj:`str`):
+                a string containing some text
+        Return:
+            A :obj:`list`:. The object returned should be a list of one list like [[{"label": 0.9939950108528137}]] containing:
+                - "label": A string representing what the label/class is. There can be multiple labels.
+                - "score": A score between 0 and 1 describing how confident the model is for this label/class.
+        """
+        if len(inputs.split()) > 1:
+            raise ValueError("Expected input is a single word")
```

Comment:
That's odd ;) but ok I guess.

Reply:
The decision at https://github.com/huggingface/huggingface_hub/issues/366 was to model this task as a text-classification and just modify the input. Maybe in the modified widget we could just enforce a one word input (cc @mishig25)

Reply:
Don't think it's worth it, just probably a little odd on the user end (I am not sure how to fix it properly, this is good enough, and would provide a good message so that's ok to me.